bentoml/_internal/yatai_client/yatai.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from typing import Optional
+from urllib.parse import urljoin
+
+import requests
+
+from .errors import YataiCliError
+from .schemas import UserSchema, BentoSchema, CreateBentoSchema, CreateBentoVersionSchema, BentoVersionSchema, \
+    OrganizationListSchema, FinishUploadBentoVersionSchema, OrganizationSchema, ModelSchema, CreateModelSchema, \
+    ModelVersionSchema, CreateModelVersionSchema, FinishUploadModelVersionSchema
+
+logger = logging.getLogger(__name__)
+
+
+class YataiClient:
+    def __init__(self, endpoint: str, api_token: str) -> None:
+        self.endpoint = endpoint
+        self.session = requests.Session()
+        self.session.headers.update({
+            'X-YATAI-API-TOKEN': api_token,
+            'Content-Type': 'application/json',
+        })
+
+    def _is_not_found(self, resp: requests.Response) -> bool:
+        # Forgive me, I don't know how to map the error returned by gorm to juju/errors
+        return resp.status_code == 400 and 'record not found' in resp.text
+
+    def _check_resp(self, resp: requests.Response) -> None:
+        if resp.status_code != 200:
+            raise YataiCliError(f'request failed with status code {resp.status_code}: {resp.text}')
+
+    def get_current_user(self) -> Optional[UserSchema]:
+        url = urljoin(self.endpoint, '/api/v1/auth/current')
+        resp = self.session.get(url)
+        if self._is_not_found(resp):
+            return None
+        self._check_resp(resp)
+        return UserSchema.from_json(resp.text)
+
+    def get_current_organization(self) -> Optional[OrganizationSchema]:
```

Comment:
also remove?

Reply:
We need this function to get the current organization from the api_token, which mainly serves to inform the user